static/js/subs.js

```diff
@@ -741,19 +740,27 @@ $(function () {
     $("#stream_creation_form").on("submit", function (e) {
         e.preventDefault();
         var stream = $.trim($("#create_stream_name").val());
-        var principals = _.map(
-            $("#stream_creation_form input:checkbox[name=user]:checked"),
-            function (elem) {
-                return $(elem).val();
-            }
-        );
-        // You are always subscribed to streams you create.
-        principals.push(page_params.email);
-        ajaxSubscribeForCreation(stream,
-            principals,
-            $('#stream_creation_form input[name=privacy]:checked').val() === "invite-only",
-            $('#announce-new-stream input').prop('checked')
+        var stream_status = compose.check_stream_existence(stream);
```

Comment:
I'd prefer to not do this synchronous query for whether the stream exists -- `compose.check_stream_existence` is slated for removal anyway.

Instead, we should just try creating the stream, and if it already exists, throw an appropriate error (I think that's what the old code did, so maybe we can just drop this section of the diff?).  Or am I missing something here?